iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPSubsystemAdd.java

```diff
@@ -194,10 +194,19 @@ public void execute(DeploymentProcessorTarget processorTarget) {
             builder.addDependency(authContextServiceName);
         }
 
+        final boolean serverRequiresSsl = IIOPRootDefinition.SERVER_REQUIRES_SSL.resolveModelAttribute(context, model).asBoolean();
+
         // inject the socket bindings that specify IIOP and IIOP/SSL ports.
         String socketBinding = props.getProperty(Constants.ORB_SOCKET_BINDING);
-        builder.addDependency(SocketBinding.JBOSS_BINDING_NAME.append(socketBinding), SocketBinding.class,
-                orbService.getIIOPSocketBindingInjector());
+        if (socketBinding != null) {
```

Comment:
Can socketBinding actually every be null? Looking at the resource definition it appears to have a default value, which means there will always be a WARN message if server-requires-ssl is set.